wikimedia/wikipedia-ios
# Remove old donor behaviors

**Phabricator:** 
https://phabricator.wikimedia.org/T349236

### Notes
This PR cleans out some old donor experience code, so now users will see the new donor experience for all valid campaigns from the config.

Note that the production [config](https://donate.wikimedia.org/wiki/MediaWiki:AppsCampaignConfig.json) is currently paused, but we can test against the staging [config](https://test.wikipedia.org/wiki/MediaWiki:AppsCampaignConfig.json). I added a valid US campaign for testing against this.

### Test Steps
1. Fresh install app on the Staging scheme in simulator.
2. Note that immediately on launch, the Settings icon shows "Donate" instead of "Support Wikipedia". This is a change, in that we no longer need the valid NL/IT campaign running in order for this path to open up.
3. Confirm that upon tapping donate you see the payment method action sheet and can tap through to those screens (note Apple Pay won't go through on Simulator).
4. Change device region to US. Background the app, foreground and pull to refresh on Explore feed. Visit EN Wiki article. Confirm a fundraising modal appears.
3. Confirm that upon tapping donate you see the payment method action sheet and can tap through to those screens.
